examples/resolvers-planetscale/grafbase/resolvers/products/all.ts

```diff
@@ -0,0 +1,81 @@
+import { connect, cast } from "@planetscale/database";
+
+const config = {
+  host: process.env.DATABASE_HOST,
+  username: process.env.DATABASE_USERNAME,
+  password: process.env.DATABASE_PASSWORD,
+};
+
+const conn = connect(config);
+
+export const options = {
+  cast(field, value) {
+    switch (field.name) {
+      case "id": {
+        return String(value);
+      }
+      case "onSale": {
+        return Boolean(value);
+      }
+      default: {
+        return cast(field, value);
+      }
+    }
+  },
+};
+
+export default async function ProductsAll(_, args) {
```

Comment:
For a tutorial, as you do queries by hand, I wouldn't bother with relay-style cursor IMHO, just offset + limit

Reply:
Happy to redo this section of the tutorial if you think it's worth it. I initially wanted to go a bit further than every other tutorial which avoids the difficult task of teaching cursor based.

Reply:
Maybe we could add 2 sub headings for adding pagination and document cursor/offset versions.